src/psyclone/tests/domain/lfric/lfric_loop_test.py

```diff
@@ -208,6 +215,30 @@ def test_mesh_name_intergrid():
     assert loops[0]._mesh_name == "mesh_field1"
 
 
+def test_lower_to_language():
+    ''' Tests that we can call lower_to_language_level a DynLoop.
+    '''
+    _, invoke = get_invoke("22.1_intergrid_restrict.f90", TEST_API, idx=0)
+
+    # Now lower the loop:
+    sched = invoke.schedule
+    loop = sched.children[2]
+    assert isinstance(loop, DynLoop)
+    sched.children[2].lower_to_language_level()
+    loop = sched.children[2]
+    assert not isinstance(loop, DynLoop)
+    assert isinstance(loop, PSyLoop)
+
+    _, invoke = get_invoke("25.0_domain.f90", TEST_API, idx=0)
+    sched = invoke.schedule
+    # Make sure we have the right node:
+    assert isinstance(sched[0], DynLoop)
+    # This call removes the loop and replaces it with the actual kernel
```

Comment:
Please extend this test to check that the loop body is copied over correctly - I think that at the moment the statements will get reversed.

Reply:
Very good catch. I've added a test that indeed showed that there was a bug, and then fixed the bug.